### DISCUSSION TITLE
[CAPI] Is there any way to fetch chunk with duckdb_fetch_chunk from duckdb_result again?

### DISCUSSION THREAD

**Discussion by suketa** [duckdb/duckdb#12442] [Q&A] [answered]
It might be related to https://github.com/duckdb/duckdb/discussions/12394, but I'd like to know about `duckdb_fetch_chunk` CAPI.

`duckdb_result_chunk_count` and `duckdb_result_get_chunk` will be deprecated by https://github.com/duckdb/duckdb/pull/12259
I think `duckdb_fetch_chunk` is provided instead.

Is there any way to fetch chunk again from the beginning after fetched all result?
Or  is there any way to rewind duckdb_result to fetch again?

## before

```C
    chunk_count = duckdb_result_chunk_count(result);

    for (chunk_idx = 0; chunk_idx < chunk_count; chunk_idx++) {
        chunk = duckdb_result_get_chunk(result, chunk_idx);
        ...
    }
    // get chunk again from the beginning.
    for (chunk_idx = 0; chunk_idx < chunk_count; chunk_idx++) {
        chunk = duckdb_result_get_chunk(result, chunk_idx);
        ...
    }
```

## after

```c
    while((chunk = duckdb_fetch_chunk(result)) != NULL) {
       ...
    }
    // try to fetch chunk again from the beginning but duckdb_fetch_chunk returns NULL.
    // Is there any way to rewind result?
    while((chunk = duckdb_fetch_chunk(result)) != NULL) {
       ...
    }
```

**Comment by Tishj** [comment, marked as answer]
The MaterializedQueryResult has that ability, it was removed(deprecated) from the generic duckdb_result API because the QueryResult interface does not support it

There will likely be a replacement method in the future to get this information specifically when dealing with a materialized query result
